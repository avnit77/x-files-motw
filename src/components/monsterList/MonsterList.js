import React from 'react';
import styles from './MonsterList.css';
import { useMonsters } from '../../hooks/motw';
import { Link } from 'react-router-dom';

const MonsterList = () => {
  const { monsterArray, page, handlePageChange } = useMonsters();

  return (
    <div className={styles.MonsterList}>
      <div className='page'>
        <button value='prev' onClick={({ target }) => handlePageChange(target.value)}>Prev</button>
        <span> Page {page} </span>
        <button value='next' onClick={({ target }) => handlePageChange(target.value)}>Next</button>
      </div>
      <ul>
        { monsterArray.length ?
          monsterArray.map(({ name, image }) => {
            return <li key={name} >
              <Link to={`/characters/${name}`} >
                <h2>{name}</h2>
                <img src={image}/>
              </Link>
            </li>;}) : null
        }
      </ul>

    </div>
  );
};

export default MonsterList;
