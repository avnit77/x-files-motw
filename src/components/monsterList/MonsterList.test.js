import React from 'react';
import { shallow } from 'enzyme';
import MonsterList from './MonsterList';

describe('MonsterList component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<MonsterList />);
    expect(wrapper).toMatchSnapshot();
  });
});
