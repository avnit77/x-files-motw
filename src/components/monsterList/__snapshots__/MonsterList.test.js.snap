// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonsterList component matches a snapshot 1`] = `
<div
  className="MonsterList"
>
  <div
    className="page"
  >
    <button
      onClick={[Function]}
      value="prev"
    >
      Prev
    </button>
    <span>
       Page 
      1
       
    </span>
    <button
      onClick={[Function]}
      value="next"
    >
      Next
    </button>
  </div>
  <ul />
</div>
`;
